src/core_init_atmosphere/mpas_init_atm_o3mr.F

```diff
@@ -0,0 +1,352 @@
+       call mpas_log_write('minval,maxval of LANDSEA = $r $r', realArgs=(/minval(maskslab), maxval(maskslab)/))
+    endif
+    deallocate(field%slab)
+    call read_next_met_field(field, istatus)
+ enddo
+ call read_met_close()
+
+ if(.not. allocated(maskslab)) then
+    call mpas_log_write('**********************************************************', messageType=MPAS_LOG_ERR)
+    call mpas_log_write('LANDSEA field not found in meteorological data file ' &
+                              //trim(config_met_prefix)//':'//config_start_time(1:13),messageType=MPAS_LOG_ERR)
+    call mpas_log_write('**********************************************************', messageType=MPAS_LOG_CRIT)
+ endif
+
+!horizontally interpolate O3 mixing ratio:
+ istatus = 0
+ call read_met_init(trim(config_met_prefix),.false.,config_start_time(1:13),istatus)
+ if(istatus /= 0) then
+    call mpas_log_write('**********************************************************', messageType=MPAS_LOG_ERR)
+    call mpas_log_write('Error opening initial meteorological data file ' &
+                              //trim(config_met_prefix)//':'//config_start_time(1:13),messageType=MPAS_LOG_ERR)
+    call mpas_log_write('**********************************************************', messageType=MPAS_LOG_CRIT)
+ endif
+ call read_next_met_field(field,istatus)
+
+ icount = 0
+ do while(istatus == 0)
+
+    interp_list(1) = FOUR_POINT
+    interp_list(2) = W_AVERAGE4
+    interp_list(3) = W_AVERAGE16
+    interp_list(4) = SEARCH
+    interp_list(5) = 0
+
+    maskval = -1.0
+    masked  = -1
+    fillval = 0.0
+    msgval  = 0.0
+
+    mask_array => landmask
+
+    if(index(field%field,'O3MR') /= 0) then
+
+       icount = icount+1
+!      call mpas_log_write('--- icount = $i',intArgs=(/icount/))
+
+       !
+       !set up projection:
+       !
+       call map_init(proj)
+   
+       if(field%iproj == PROJ_LATLON) then
+          call map_set(PROJ_LATLON,proj, &
+                       latinc = real(field%deltalat,RKIND), &
+                       loninc = real(field%deltalon,RKIND), &
+                       knowni = 1.0_RKIND, &
+                       knownj = 1.0_RKIND, &
+                       lat1   = real(field%startlat,RKIND), &
+                       lon1   = real(field%startlon,RKIND))
+       elseif(field%iproj == PROJ_GAUSS) then
+          call map_set(PROJ_GAUSS,proj, &
+                       nlat = nint(field%deltalat), &
+                       loninc = 360.0_RKIND / real(field%nx,RKIND), &
+                       lat1 = real(field%startlat,RKIND), &
+                       lon1 = real(field%startlon,RKIND))
+       endif
+
+       !
+       !horizontally interpolate field at level k:
+       !
+
+       do k = 1,nFGLevels
+          if(field%xlvl == p_fg(k,1)) exit
+       enddo
+       call mpas_log_write('Interpolating O3MR at $r $r $i',realArgs=(/field%xlvl,p_fg(k,1)/), &
+                           intArgs=(/k/))
```

Comment:
GNU compiler complains about R4 and R8  data format on Cheyenne.  A small bug here: 
-   call mpas_log_write('Interpolating O3MR at $r $r $i',realArgs=(/field%xlvl,p_fg(k,1)/), &
-                           intArgs=(/k/))
+       call mpas_log_write('Interpolating O3MR at $r $r $i',&
+         realArgs=(/real(field%xlvl,RKIND),p_fg(k,1)/), intArgs=(/k/))
Otherwise all builds are successful.

Reply:
I agree that an explicit type-conversion to `real(kind=RKIND)` is needed for `field%xlvl`, which is defined as `real(kind=4)` in [mpas_init_atm_read_met.F](https://github.com/MPAS-Dev/MPAS-Model/blob/v7.3/src/core_init_atmosphere/mpas_init_atm_read_met.F#L27).